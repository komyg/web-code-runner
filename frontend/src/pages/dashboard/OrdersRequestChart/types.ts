export interface SummarizedOrderRequestData {
  startDate: Date;
  endDate: Date;
  totalRequests: number;
  successfulRequests: number; // 2XX
  failedRequests: number; // 4XX
  serverErrorRequests: number; // 500
  badGatewayRequests: number; // 502
  serviceUnavailableRequests: number; // 503
  gatewayTimeoutRequests: number; // 504
}

export interface OrderStatistics extends SummarizedOrderRequestData {
  time: string;
}
