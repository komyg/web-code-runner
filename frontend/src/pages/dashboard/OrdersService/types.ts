export interface ServiceData {
  serviceId: string;
  serviceName: string;
  numReplicas: number;
  status: string;
}
