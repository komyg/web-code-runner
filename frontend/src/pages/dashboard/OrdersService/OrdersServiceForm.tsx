import { Box, Button, TextField } from '@mui/material';
import { useFormik } from 'formik';
import * as yup from 'yup';

interface Props {
  numReplicas: number;
}

const validationSchema = yup.object({
  numReplicas: yup
    .number()
    .required('Number of replicas is required')
    .min(1, 'Number of replicas must be greater than 0'),
});

export function OrdersServiceForm({ numReplicas }: Props) {
  const formik = useFormik({
    initialValues: {
      numReplicas,
    },
    validationSchema,
    onSubmit: (values) => {
      console.log(values);
    },
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <Box display="flex" flexDirection="column" gap={2}>
        <TextField
          variant="standard"
          id="numReplicas"
          name="numReplicas"
          label="Number of Replicas"
          value={formik.values.numReplicas}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={
            formik.touched.numReplicas && Boolean(formik.errors.numReplicas)
          }
          helperText={formik.touched.numReplicas && formik.errors.numReplicas}
        />
        <Button type="submit" variant="contained">
          Update
        </Button>
      </Box>
    </form>
  );
}
