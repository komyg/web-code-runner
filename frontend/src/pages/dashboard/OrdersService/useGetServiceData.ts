import { useEffect, useState } from 'react';
import { ServiceData } from './types';

export function useGetServiceData(serviceId: string) {
  const [data, setData] = useState<ServiceData>();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const loadOrdersServiceData = async () => {
      setLoading(true);
      const url = `${import.meta.env.VITE_API_URL}/service/${serviceId}`;
      const response = await fetch(url);
      const data = await response.json();
      setData(data);
      setLoading(false);
    };
    loadOrdersServiceData();
  }, []);

  return { data, loading };
}
