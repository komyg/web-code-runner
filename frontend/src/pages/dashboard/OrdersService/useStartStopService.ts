import { useState } from 'react';

export function useStartService() {
  const [loading, setLoading] = useState(false);

  const startService = async (serviceId: string) => {
    try {
      setLoading(true);
      const url = `${import.meta.env.VITE_API_URL}/service/${serviceId}/start`;
      await fetch(url, {
        method: 'POST',
      });
    } catch (error) {
      console.error('Error updating service', error);
    }

    setLoading(false);
  };

  return { loading, startService };
}

export function useStopService() {
  const [loading, setLoading] = useState(false);

  const stopService = async (serviceId: string, deploymentId: string) => {
    try {
      setLoading(true);
      const url = `${import.meta.env.VITE_API_URL}/service/${serviceId}/stop`;
      await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ deploymentId }),
      });
    } catch (error) {
      console.error('Error updating service', error);
    }

    setLoading(false);
  };

  return { loading, stopService };
}
