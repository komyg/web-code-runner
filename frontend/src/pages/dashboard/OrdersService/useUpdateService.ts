import { useState } from 'react';

interface UpdateServiceData {
  numReplicas: number;
}

export function useUpdateService() {
  const [loading, setLoading] = useState(false);

  const updateService = async (serviceId: string, data: UpdateServiceData) => {
    try {
      setLoading(true);
      const url = `${import.meta.env.VITE_API_URL}/service/${serviceId}`;
      await fetch(url, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(data),
      });
    } catch (error) {
      console.error('Error updating service', error);
    }

    setLoading(false);
  };

  return { loading, updateService };
}
