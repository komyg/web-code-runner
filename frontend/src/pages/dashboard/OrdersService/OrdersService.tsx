import { Box, Typography } from '@mui/material';
import { ServiceData } from './types';
import { OrdersServiceForm } from './OrdersServiceForm';

interface Props {
  serviceData: ServiceData;
}

export function OrdersService({ serviceData }: Props) {
  return (
    <Box display="flex" flexDirection="column" gap={2}>
      <Typography variant="h6">{serviceData.serviceName}</Typography>
      <Typography>Service ID: {serviceData.serviceId}</Typography>
      <Typography>Status: {serviceData.status}</Typography>
      <OrdersServiceForm numReplicas={serviceData.numReplicas} />
    </Box>
  );
}
