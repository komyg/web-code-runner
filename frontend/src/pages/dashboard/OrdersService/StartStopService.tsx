import { Box, Button } from '@mui/material';
import { useStartService, useStopService } from './useStartStopService';

interface Props {
  serviceId: string;
  serviceStatus: string;
  deploymentId: string;
}

export function StartStopService({
  serviceId,
  deploymentId,
  serviceStatus,
}: Props) {
  const { loading: startServiceLoading, startService } = useStartService();
  const { loading: stopServiceLoading, stopService } = useStopService();
  const loading = startServiceLoading || stopServiceLoading;

  const startButton = (
    <Button
      variant="contained"
      color="secondary"
      fullWidth
      disabled={loading}
      onClick={() => startService(serviceId)}
    >
      Start
    </Button>
  );
  const stopButton = (
    <Button
      variant="contained"
      color="secondary"
      fullWidth
      disabled={loading}
      onClick={() => stopService(serviceId, deploymentId)}
    >
      Stop
    </Button>
  );

  return (
    <Box display="flex" justifyContent="space-around">
      {serviceStatus === 'SUCCESS' ? stopButton : startButton}
    </Box>
  );
}
