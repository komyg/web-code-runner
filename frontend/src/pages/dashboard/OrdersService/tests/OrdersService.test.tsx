import { render } from '@testing-library/react';
import { expect, it } from 'vitest';
import { OrdersService } from '../OrdersService';
import { ServiceData } from '../types';

it('should match the snapshot', () => {
  const fakeServiceData: ServiceData = {
    serviceId: 'fake-service-id',
    serviceName: 'fake-service',
    numReplicas: 1,
    status: 'SUCCESS',
    deploymentId: 'fake-deployment-id',
  };
  const { asFragment } = render(
    <OrdersService serviceData={fakeServiceData} />
  );
  expect(asFragment()).toMatchSnapshot();
});
