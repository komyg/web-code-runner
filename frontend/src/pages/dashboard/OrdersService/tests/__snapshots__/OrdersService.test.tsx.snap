// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-yd8sa2"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
    >
      fake-service
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Service ID: fake-service-id
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Status: SUCCESS
    </p>
    <form>
      <div
        class="MuiBox-root css-yd8sa2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-nmluci-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="numReplicas"
            id="numReplicas-label"
          >
            Number of Replicas
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-r5in8e-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1o3an4k-MuiInputBase-input-MuiInput-input"
              id="numReplicas"
              name="numReplicas"
              type="text"
              value="1"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
    <div
      class="MuiBox-root css-1bnxa12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-fullWidth css-od1n8g-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Stop
      </button>
    </div>
  </div>
</DocumentFragment>
`;
